packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -514,23 +615,6 @@ function patchElementPropsAndAttrsAndRefs(vnode: VBaseElement, renderer: Rendere
     applyRefs(vnode, vnode.owner);
 }
 
-function hasCorrectNodeType<T extends Node>(
```

Comment:
replaced with dedicated guards (isTypeElement, etc) for type inference (avoids [forcing the type](https://github.com/salesforce/lwc/pull/5086/files#diff-a9673af211cdbdb5096c7ee4533b234a8bf90e9271341048b42ffcd6c0e143ceL861) and then providing a separate type parameter)